### ISSUE TITLE
defaultPosture configuration is not being respect, audit is configured but it's generating block events.

### ISSUE THREAD

**Issue by kaiquerass** [kubearmor/KubeArmor#812]
## Bug Report

**General Information**

- GKE 1.22

**To Reproduce**

1. Include this configuration in the yaml file:
```
    spec:
      containers:
      - args:
        - -defaultFilePosture=audit
        - -hostDefaultFilePosture=audit
```

**Expected behavior**
defaultPosture configured to audit.

Pod's logs shows the configuration is corrected and is applied: 
```
2022-08-11 19:18:47.523563	INFO	Configuration [{Cluster:default Host:xxxxx GRPC:32767 LogPath:/tmp/kubearmor.log SELinuxProfileDir:/tmp/kubearmor.selinux CRISocket: Visibility:process,file,network,capabilities HostVisibility:default Policy:true HostPolicy:false KVMAgent:false K8sEnv:true DefaultFilePosture:audit DefaultNetworkPosture:block DefaultCapabilitiesPosture:block HostDefaultFilePosture:audit HostDefaultNetworkPosture:block HostDefaultCapabilitiesPosture:block CoverageTest:false}]
2022-08-11 19:18:47.523585	INFO	Final Configuration [{Cluster:default Host:xxxxxxxxxxx GRPC:32767 LogPath:/tmp/kubearmor.log SELinuxProfileDir:/tmp/kubearmor.selinux CRISocket: Visibility:process,file,network,capabilities HostVisibility:none Policy:true HostPolicy:false KVMAgent:false K8sEnv:true DefaultFilePosture:audit DefaultNetworkPosture:block DefaultCapabilitiesPosture:block HostDefaultFilePosture:audit HostDefaultNetworkPosture:block HostDefaultCapabilitiesPosture:block CoverageTest:false}
```

I expected to see only audit events, but I'm seeing many block events with defaultPosture policy, as shown below.

```
== Alert / 2022-08-11 19:31:47.544754 ==
ClusterName: default
HostName: XXXXX
NamespaceName: XXXX
PodName: XXXXX
Labels: XXXXX
ContainerName: XXXX
ContainerID: XXXXX
ContainerImage: XXXXX
Type: MatchedPolicy
PolicyName: DefaultPosture
Source: /usr/local/bin/python
Resource: /usr/local/lib/python/XXXXXXXX
Operation: File
Action: Block
Data: syscall=SYS_OPENAT fd=-100 flags=O_WRONLY|O_CREAT|O_EXCL|O_CLOEXEC
Enforcer: AppArmor
Result: Permission denied
HostPID: XXX
HostPPID: XXX
PID: XXX
PPID: XXX
ParentProcessName: /usr/local/bin/python
ProcessName: /usr/local/bin/python
UID: XXXX

```

**Comment by daemon1024** [comment]
Hey @kaiquerass,  Thank You for reporting this.

Can you share the policies applied in the cluster? Also if possible sharing the dump created by [`karmor sysdump`](https://github.com/kubearmor/kubearmor-client) if you are comfortable doing so.

**Comment by kaiquerass** [comment]
sorry, I fogot to mention this.

```
apiVersion: security.kubearmor.com/v1
kind: KubeArmorPolicy
metadata:
  name: teste
  namespace: echo
spec:
  severity: 10
  message: "a critical file was accessed"
  tags:
  - WARNING
  selector:
    matchLabels:
     app: echo
  file:
    matchPaths:
    - path: /etc/passwd
    - path: /secret.txt
    - path: /credentials/password
  action:
    Block
```

**Comment by daemon1024** [comment]
@kaiquerass 
The defaultPosture you used in the configuration only is relevant when you have any `Allow` Policy or the rules are whitelist and not blacklist. This was introduced because Whitelists are difficult to handle and can cause implicit denies.
More about it at https://github.com/kubearmor/KubeArmor/blob/main/getting-started/default_posture.md

But as the policy mentions an explicit block, `-defaultPostur=audit` isn't relevant in this case since there's an explicit mention to Block it. You can change the Block to Audit in the policy itself and it will serve the purpose.

That said, I see `DefaultPosture` mentioned in the telemetry you attached with Resources not relevant to the policy. This happens when there's a denial due to systems outside of KubeArmor. You might have some other systems in place like AppArmor profiles blocking the resource you are trying to execute. 

I realise these 2 `DefaultPosture`s might be confusing and we are trying to create a distinction between the 2 cases or document this scenario by the next release. Any feedback appreciated, happy to answer followup questions/ debug things more here.

**Comment by kaiquerass** [comment]
Just to check if I understood. KubeArmor will send an event, even if it's not KubeArmor that is blocking. 
But it's weird because we don't have any system like AppArmor. KubeArmor is the only one.

**Comment by daemon1024** [comment]
> Just to check if I understood. KubeArmor will send an event, even if it's not KubeArmor that is blocking. 

Yes 😅 

> But it's weird because we don't have any system like AppArmor. KubeArmor is the only one.

Just to confirm do you see these alerts even if no policy is applied?

**Comment by kaiquerass** [comment]
I couldn't see the same alert when I deleted the policy, but I see other alerts. This means there is something somehow blocking outside KubeArmor. 


It would be awesome to have a different type on this alerts.
The grafana dashboard is useless since I can't filter alerts per policy. 
The only metric useful is `kubearmor_alerts_with_policy_total`, because I can filter only my policies.
Maybe is important is to add the policy name in other metrics, that way I could see alerts per namespace, filtering to exclude defaultPosture. For example: `max(kubearmor_alerts_in_namespace_total{policy_name!="defaultPosture"}) by (NamespaceName)`.

The way that is implemented, is not functional, the dashboard has lots of alerts per namespaces, but actually, just 5 events from 2 namespaces are real alerts, the ones that I want to see on dashboard. Only KubeArmor blocked actions from my policies.

**Comment by kaiquerass** [comment]
hey @daemon1024, just some updates: 

actually, this defaultPosture shown in events, is KubeArmor blocking the access. Today we had an incident with many pods/deployments being blocked in the startup. We had istio deployment being blocked, and this has and direct impact in our backend applications, because there's an istio init container running on them. Blocking istio init container, blocked our applications. 
When I deleted the KubeArmor annotations from istio deployment, it worked as expected. 

We had other deployments being blocked, like prometheus, grafana... And the same happened when I deleted the annotations, everything went up again.

**Comment by nyrahul** [comment]
> hey @daemon1024, just some updates:
> 
> actually, this defaultPosture shown in events, is KubeArmor blocking the access. Today we had an incident with many pods/deployments being blocked in the startup. We had istio deployment being blocked, and this has and direct impact in our backend applications, because there's an istio init container running on them. Blocking istio init container, blocked our applications. When I deleted the KubeArmor annotations from istio deployment, it worked as expected.
> 
> We had other deployments being blocked, like prometheus, grafana... And the same happened when I deleted the annotations, everything went up again.

Hey @kaiquerass , is this problem reproducible? Can you specify the steps to reproduce? We will try it out and check what happens.

**Comment by kaiquerass** [comment]
I'm not sure if it's reproducible, because I installed KubeArmor using default settings, and it's a problem involving many other technologies. But, these are the affected technologies and the versions, respectively. 

Istio: 1.9.8
Grafana: 8.5.6 (623255cf2d)
Prometheus: 2.33.3
Thanos: 0.25.0

The problem:
First, we've seen some istio sidecars being blocked. After a while, the istiod deployment was blocked.
Grafana and prometheus deployments was blocked too.

**Comment by nyrahul** [comment]
@kaiquerass , based on the slack conversation, this issue seems to be solved by #902 ... Feel free to open it up if you think otherwise.
